### PR TITLE
 Add instrumentation and tests for Grizzly 2.x Http Server

### PR DESCRIPTION
**Pull request by tylerbenson**

### CODE REVIEW

File: dd-java-agent/instrumentation/grizzly-2/src/main/java/datadog/trace/instrumentation/grizzly/GrizzlyDecorator.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  @Override
+  protected Integer status(final Response containerResponse) {
+    return containerResponse.getStatus();
+  }
+
+  @Override
+  protected String[] instrumentationNames() {
+    return new String[] {"jersey"};
```
Comment (randomanderson) [inline dd-java-agent/instrumentation/grizzly-2/src/main/java/datadog/trace/instrumentation/grizzly/GrizzlyDecorator.java]: jersey -> grizzly

File: dd-java-agent/instrumentation/grizzly-2/src/main/java/datadog/trace/instrumentation/grizzly/GrizzlyDecorator.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  @Override
+  protected String component() {
+    return "jersey";
```
Comment (randomanderson) [inline dd-java-agent/instrumentation/grizzly-2/src/main/java/datadog/trace/instrumentation/grizzly/GrizzlyDecorator.java]: jersey -> grizzly

File: dd-java-agent/instrumentation/grizzly-2/src/test/groovy/GrizzlyAsyncTest.groovy [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  @Override
+  HttpServer startServer(int port) {
+    ResourceConfig rc = new ResourceConfig()
+    rc.register(SimpleExceptionMapper)
+    rc.register(ServiceResource)
+    GrizzlyHttpServerFactory.createHttpServer(new URI("http://localhost:$port"), rc)
+  }
+
+  @Path("/")
+  static class ServiceResource {
```
Comment (randomanderson) [inline dd-java-agent/instrumentation/grizzly-2/src/test/groovy/GrizzlyAsyncTest.groovy]: I think it's slightly confusing to have the same private class name in a class and superclass.  Maybe `ServiceResource` -> `AsyncServiceResource`?

File: dd-java-agent/instrumentation/grizzly-2/src/main/java/datadog/trace/instrumentation/grizzly/GrizzlyDecorator.java (hunk 1 above)
Comment (tylerbenson) [reply dd-java-agent/instrumentation/grizzly-2/src/main/java/datadog/trace/instrumentation/grizzly/GrizzlyDecorator.java]: Good catch.

**Pull request merged**